d3a_api_client/redis_client_base.py

```diff
@@ -313,4 +307,12 @@ def on_trade(self, trade_info):
         pass
 
     def on_finish(self, finish_info):
-        pass
\ No newline at end of file
+        pass
+
+    def add_to_batch(self, area_uuid: str, ):
+        command = ClientCommand(area_uuid=area_uuid)
+        self.commands_buffer.append(command)
+        return command
```

Comment:
This makes adding to commands_buffer like this 
`device.add_to_batch(area_uuid).offer(energy, price)`
or 
`device.add_to_batch(area_uuid).delete_offer(offer_id)`

Reply:
Why do you need the area_uuid explicitly ? This should be part of the class already (self.area_id should give you that exact information)

Reply:
I could not understand the reason for explicitly get it from the event object instead of from self in [here](https://github.com/gridsingularity/d3a-api-client/blob/feature/D3ASIM-3213/tests/test_create_aggregator.py#L35)
In tests the area_uuid is being passed explicitly

Reply:
I also find this approach a bit more difficult for the end user than it should be. Instead of exposing a function that returns a ClientCommand object, which in turn can be used to call offer, delete_offer etc. , I think it would be much easier to either have dedicated methods on the RedisClient (similar to what we have already with the bid_energy, offer_energy), or, if you want to keep the existing structure, to expose `add_to_batch` as a property/class member and not as a function.

Reply:
The second option is what I had in mind , imo it would make the code cleaner 
anw concerning the `area_uuid` point , can I safely use the `self.area_id` ? and just to note it looks like this variable refers to the name of an area not to its ID

Reply:
OK, now I see the point of confusion. The batch_command is only part of the aggregator api, and not part of the device api. An aggregator is an entity that is able to control multiple devices at once. That's why you actually need the area_uuid argument, because the user indeed needs to define in which area should the command be sent to. 
Taking this into account, I think that the best solution would be to add it as an extra argument to each method. That IMO would make the api easier to use.